framework/PostProcessors/BasicStatistics.py

```diff
@@ -230,21 +207,15 @@ def initialize(self, runInfo, inputs, initDict):
     """
     #construct a list of all the parameters that have requested values into self.allUsedParams
     self.allUsedParams = set()
-    #first collect parameters for which scalar values were requested
-    for scalar in self.scalarVals:
-      if scalar in self.toDo.keys():
-        #special treatment of percentile since the user can specify the percents directly
-        if scalar == 'percentile':
-          for pct,targs in self.toDo[scalar].items():
-            self.allUsedParams.update(targs)
-        else:
-          self.allUsedParams.update(self.toDo[scalar])
-    #second collect parameters for which matrix values were requested, either as targets or features
-    for vector in self.vectorVals:
-      if vector in self.toDo.keys():
-        for entry in self.toDo[vector]:
+    for metricName in self.scalarVals + self.vectorVals:
+      if metricName in self.toDo.keys():
+        for entry in self.toDo[metricName]:
           self.allUsedParams.update(entry['targets'])
-          self.allUsedParams.update(entry['features'])
+          try:
+            self.allUsedParams.update(entry['features'])
+          except KeyError:
+            pass
```

Comment:
Con you explain this a bit better?

Reply:
Yes, for scalar metric, we store the variable as targets only, for matrix metric, we store the variables as targets and features, respectively.